routes/routes.js

```diff
@@ -195,54 +197,6 @@ router.post(
     }
 );
 
-router.post(
-  "/user/login",
-  [
-    check("email", "Please enter a valid email").isEmail(),
-    check("password", "Please enter a valid password").isLength({
-      min: 6
-    })
-  ],
-  async (req, res) => {
-    const errors = validationResult(req);
-
-    if (!errors.isEmpty()) {
-      return res.status(400).json({
-        errors: errors.array()
-      });
-    }
-
-    const email = req.body.email,
-          password= req.body.password
-    try {
-      let user = await User.findOne({
-        email
-      });
-      if (!user)
-        return res.status(400).json({
-          message: "User Not Exist"
-        });
-
-      const isMatch = await bcrypt.compare(password, user.password);
-      if (!isMatch){
-        return res.status(400).json({
-          message: "Incorrect Password !"
-        });
-      }
-      req.session.logged = true;
-      res.render('rosterview/selectRoster',{
-        viewTitle: 'Select a date',
-        item: user.fname,
-        user_id: user.id
-      })
-      
-    } catch (e) {
-      console.error(e);
-      res.status(500).json({
-        message: "Server Error"
-      });
-    }
-  }
-);
+router.post("/user/login", authController.login);
```

Comment:
A common pattern you will see is MVC or model view controller, most larger frameworks that favour convention over configuration will follow this pattern eg Rails, Laravel, Django (well Django is technically model view template 🤷 ). Express is very unopinionated in this regard but its a great habit to get into, I think it makes things cleaner and the code more organised. Helps when navigating a larger code base.

So here you can see clearly the route and the controller method thats passed in as a handler.

Reply:
Also no need to validate the email here anymore or the password, the browser will do that for you and validate the length of the password on login as its already been created 😄